Keep serial provided in the irrd_load_database command

Justin-APNIC opened the pull request:
Fix for https://github.com/irrdnet/irrd/issues/809

Justin-APNIC committed:
use initial serial when journal table is empty

mxsasha commented:
I was thinking we can just change the query to `COALESCE(MAX(serial_nrtm), serial_newest_seen, 0) + 1`? So if there are entries in the journal, do the usual, if not, start from serial_newest_seen, if not, start from 0 (+1 in all cases).

Justin-APNIC committed:
Refactor codes and add tests

Justin-APNIC commented:
> I was thinking we can just change the query to `COALESCE(MAX(serial_nrtm), serial_newest_seen, 0) + 1`? So if there are entries in the journal, do the usual, if not, start from serial_newest_seen, if not, start from 0 (+1 in all cases).

Thanks for looking into it. I refactored the code to use a single query.
Also, tested on my local env by loading a initial dump with an initial serial and sending a update request after that. The records in db look good.

```
irrd=# select * from database_status where source='JPNIC';
-[ RECORD 1 ]---------+-------------------------------------
pk                    | 485a9a71-b37d-4134-b18d-a67bc0cb943e
source                | JPNIC
serial_oldest_seen    | 57291985
serial_newest_seen    | 57291986
serial_oldest_journal | 57291986
serial_newest_journal | 57291986
serial_last_export    |
force_reload          | f
last_error            |
last_error_timestamp  |
created               | 2024-01-17 03:43:08.597081+00
updated               | 2024-01-17 04:00:50.27101+00
serial_newest_mirror  |
synchronised_serials  | f
```

```
irrd=# select * from rpsl_database_journal where source='JPNIC';
-[ RECORD 1 ]-+--------------------------------------------------
pk            | 251f98ab-373d-483c-8878-be6b5696d5aa
rpsl_pk       | DUMMY-JP
source        | JPNIC
serial_nrtm   | 57291986
operation     | add_or_update
object_class  | person
object_text   | person:         Dummy JPNIC Host Master          +
              | nic-hdl:        DUMMY-JP                         +
              | e-mail:         hostmaster@nic.ad.jp             +
              | address:        Chiyoda-ku, Tokyo 101-0047, Japan+
              | phone:          +81-3-5297-2311                  +
              | mnt-by:         MAINT-JPNIC                      +
              | source:         JPNIC                            +
              | last-modified:  2024-01-17T03:47:13Z             +
              |
timestamp     | 2024-01-17 04:00:50.247865+00
origin        | auth_change
serial_global | 1000009
```

If no serial provided in the `irrd_load_database` command, the `serial_nrtm` will start from 1.
